### PR TITLE
Dec endowing

### PR DESCRIPTION
**Pull request by elcerdo**
change DEC calculus internal data structure.
store both primal and dual sizes instead of ratio.

### CODE REVIEW

**Comment by elcerdo** [comment]
which syntax is the correct one for a static constructor
thisOne();
or 
ThisOne();
??

**Comment by dcoeurjo** [comment]
All classe names need to start by a Capital Letter. So does its constructor (ThisOne())

and ```thisOne()``` for a static method.

**Comment by dcoeurjo** [comment]
Ready to review ?

**Comment by elcerdo** [comment]
ready to review.
this branch and a fresh clone of dgtaltools should fix ubuntu compilation.

File: src/DGtal/dec/DiscreteExteriorCalculus.h [hunk 1]
```diff
@@ -296,15 +296,26 @@ namespace DGtal
...
+    bool
+    insertSCell(const SCell& signed_cell);
+
+    /**
      * Manually insert signed cell into calculus.
      * Should call updateIndexes() when structure modification is finished.
      * Be sure to insert all adjacent lower order primal cells.
      * @param signed_cell the signed cell to be inserted.
-     * @param size_ratio ratio of dual cell size over primal cell size.
+     * @param primal_size size (length, area, volume, ...) of primal cell.
```
Comment (dcoeurjo) [inline src/DGtal/dec/DiscreteExteriorCalculus.h]: you should clearly say that for 0-cell, measure or size must be one (asserted in the code)

Comment (dcoeurjo) [inline src/DGtal/dec/DiscreteExteriorCalculus.h]: Since the user prescribes now sizes instead of ratio, should the doc be updated ?

**Comment by elcerdo** [comment]
doc and style fixed!!

**Comment by dcoeurjo** [comment]
ok thanks.
Could you please add a changelog entry describing the changes ?

**Comment by elcerdo** [comment]
changelog updated

File: ChangeLog.md [hunk 2]
```diff
@@ -21,6 +21,13 @@
    [#1076](https://github.com/DGtal-team/DGtal/pull/1076))
 
 ## Changes
+- *DEC Package*
+ - DiscreteExteriorCalculus holds both primal and dual size.
```
Comment (dcoeurjo) [inline ChangeLog.md]: primal and dual size of each cell.

**Comment by dcoeurjo** [comment]
Great, thanks

**Pull request merged**